packages/preview-middleware-client/src/flp/enableFakeConnector.ts

```diff
@@ -0,0 +1,120 @@
+import LrepConnector from 'sap/ui/fl/LrepConnector';
+import FakeLrepConnector from 'sap/ui/fl/FakeLrepConnector';
+import type { FlexSettings } from 'sap/ui/rta/RuntimeAuthoring';
+
+interface FlexChange {
+    [key: string]: string | object | undefined;
+    support: {
+        generator?: string;
+    };
+}
+
+interface FetchedChanges {
+    [key: string]: FlexChange;
+}
+
+interface LoadChangesResult {
+    changes: {
+        loadModules: boolean;
+        changes: FlexChange[];
+        settings: {
+            [key: string]: string | boolean | undefined;
+        };
+    };
+    componentClassName: string;
+    etag: string;
+    loadModules: boolean;
+    messagebundle: string | undefined;
+}
+
+const path = '/preview/api/changes';
+
+/**
+ * Retrieves Flex settings from a 'sap-ui-bootstrap' element's data attribute.
+ * Parses the 'data-open-ux-preview-flex-settings' attribute as JSON.
+ *
+ * @returns {FlexSettings | undefined} The parsed Flex settings if available, otherwise undefined.
+ */
+function getFlexSettings(): FlexSettings | undefined {
+    let result;
+    const bootstrapConfig = document.getElementById('sap-ui-bootstrap');
+    const flexSetting = bootstrapConfig?.getAttribute('data-open-ux-preview-flex-settings');
+    if (flexSetting) {
+        result = JSON.parse(flexSetting);
+    }
+    return result;
+}
```

Comment:
We can move this in a central place. It is used by by both connectors now.

Reply:
Fixed in: https://github.com/SAP/open-ux-tools/commit/df40622768f8ed2de9899577e0a7220576cb0d9b